aws/table_aws_organizations_account.go

```diff
@@ -129,92 +154,68 @@ func listOrganizationsAccounts(ctx context.Context, d *plugin.QueryData, _ *plug
 	if d.QueryContext.Limit != nil {
 		limit := int32(*d.QueryContext.Limit)
 		if limit < maxItems {
-			if limit < 1 {
-				maxItems = int32(1)
-			} else {
-				maxItems = int32(limit)
-			}
+			maxItems = int32(limit)
 		}
 	}
 
+	var parents []types.OrganizationalUnit
+
 	// Lists the accounts in an organization that are contained by the specified target root or organizational unit (OU).
 	// If you specify the root, you get a list of all the accounts that aren't in any OU.
 	// If you specify an OU, you get a list of all the accounts in only that OU and not in any child OUs.
-	if parentId != "" {
-		op, err := listOrganizationsAccountsForParent(ctx, d, svc, maxItems, &organizations.ListAccountsForParentInput{
-			ParentId:   &parentId,
-			MaxResults: &maxItems,
-		})
+	if d.EqualsQualString("parent_id") != "" {
+		parents = []types.OrganizationalUnit{{
+			Id: aws.String(d.EqualsQualString("parent_id")),
+		}}
+	}
+
+	// Restrict listing of parents when "parent_id" is provided.
+	if !(len(parents) > 0) {
+		// Call the recursive function to list all nested OUs
```

Comment:
Is it a recursive function?

Reply:
Yes, It is recursive, we already do it in the table aws_organizations_organizational_unit.